pkg/apis/resource/install/install.go

```diff
@@ -25,6 +25,7 @@ import (
 	"k8s.io/kubernetes/pkg/apis/resource"
 	"k8s.io/kubernetes/pkg/apis/resource/v1alpha3"
 	"k8s.io/kubernetes/pkg/apis/resource/v1beta1"
+	"k8s.io/kubernetes/pkg/apis/resource/v1beta2"
```

Comment:
> This PR adds the v1beta2 API, but doesn't update the scheduler/allocator or the e2e tests to use the new API.

I agree with that decision, primarily because it avoids one potential pitfall: users who continue to enable only the v1beta1 API in their cluster would end up with an unusable cluster if the control plane components required v1beta2.

Let's start planning the v1beta1 removal. Can you:
- add a remark to the release note: `DRA: the v1beta1 is deprecated. It will be removed in 1.36. Use v1beta2 instead.`
- create an issue as reminder, which includes rewriting the control plane components
- add the `/kind deprecation` (not doing it now myself in case that we want to discuss further)

Reply:
> I'm also wondering if we should update the e2e tests. It seems like maybe we should have tests both for v1beta1 and v1beta2, but that ends up being a lot of tests.

Let's do a few E2E test which uses v1beta2:
- one with `exactly`
- one with `firstAvailable`

Remember to add v1beta2 to `test/e2e/dra/kind.yaml`, otherwise those tests will fail in `pull-kubernetes-kind-dra`.

Reply:
Created issue https://github.com/kubernetes/kubernetes/issues/129891 and added a release note and the deprecation label to the PR.

Also added e2e tests for the `v1beta2` API.